substrate/enum_utils

```diff
@@ -21,75 +21,75 @@ namespace substrate
 
 	template<class T> struct enumPair_t final
 	{
-		using value_type = T;
 	private:
-		T _value;
-		const char* _name;
+		T _value{};
+		const char *_name{};
+
 	public:
-		constexpr enumPair_t() noexcept : _value{}, _name{} { }
+		constexpr enumPair_t() noexcept = default;
 		constexpr enumPair_t(T value, const char* name) noexcept : _value{value}, _name{name} { }
 
+		using value_type = T;
+
 		void value(const T value) noexcept { _value = value; }
 		SUBSTRATE_NO_DISCARD(T value() const noexcept) { return _value; }
 
-		void name(const char* name) noexcept { _name = name; }
-		SUBSTRATE_NO_DISCARD(const char* name() const noexcept) { return _name; }
+		void name(const char *name) noexcept { _name = name; }
+		SUBSTRATE_NO_DISCARD(const char *name() const noexcept) { return _name; }
 	};
 
 	/* Extract a collection of flags set in a field */
 	/* This is kind of expensive run-time wise, being at leas O(n+1) but *shrug* */
 	template<typename T, typename A> enable_if_t<std::is_enum<T>::value, std::vector<T>>
-		extract_flags(T flags, A& enum_table)
+		extract_flags(T flags, const A &enumTable)
 	{
-		std::vector<T> _found_flags;
-		using ut = typename substrate::underlying_type_t<T>;
-		for (auto flag : enum_table)
+		std::vector<T> foundFlags{};
```

Comment:
nitpick: You can use an upper bound for the vector capacity (enumTable.size, if available).